source/Launchpad-Commands/TCPCommandServer.class.st

```diff
@@ -0,0 +1,81 @@
+Class {
+	#name : #TCPCommandServer,
+	#superclass : #CommandServer,
+	#instVars : [
+		'listeningPort',
+		'commmands',
+		'connectionSocket',
+		'serverProcess'
+	],
+	#category : #'Launchpad-Commands'
+}
+
+{ #category : #'instance creation' }
+TCPCommandServer class >> listeningOn: port [
+
+	^ self new initializeListeningOn: port
+]
+
+{ #category : #private }
+TCPCommandServer >> initializeConnectionSocket [
+
+	connectionSocket := Socket newTCP
+]
+
+{ #category : #initialize }
+TCPCommandServer >> initializeListeningOn: port [
+
+	listeningPort := port.
+	commmands := Dictionary new.
+	self initializeConnectionSocket
+]
+
+{ #category : #accessing }
+TCPCommandServer >> listeningPort [
+
+	^ listeningPort
+]
+
+{ #category : #registering }
+TCPCommandServer >> registerCommandNamed: commandName executing: aBlock [
+
+	commmands at: commandName put: aBlock
+]
+
+{ #category : #private }
+TCPCommandServer >> serverLoop [
+
+	^ [ 
+	  ( connectionSocket waitForAcceptFor: 60 ) ifNotNil: [ :socket | 
```

Comment:
A timeout for accept is not needed afaik
```suggestion
	  connectionSocket accept ifNotNil: [ :socket | 
```

Reply:
This change blocks the VM until a connection is made to the socket, where the previous code is non-blocking until a connection is made. So I'm in favor of keeping the old behavior, the timeout is a non-issue because its inside a never-ending loop so once the timeout is reached it will wait again.

Reply:
I assumed accept did not lock the vm, only the thread, leave it as is then.